### DISCUSSION TITLE
Creating a basic, standalone node with zero maintenance

### DISCUSSION THREAD

**Discussion by stanhamil** [meshtastic/Discussions#127] [General]
I have a family member living in a semi-high rise block in central London and figure the location is ideal for a Meshtastic node.

What’s the simplest solution for such a node? I won’t be able to access it physically very often (if at all) but can maintain it via an admin key if the LoRA device has WiFi capabilities (I’ve set up my own MQTT server).

Ideally, the node will be in a case on a windowsill and I’d like a small solar panel to be attached to it. Even better is if there’s such a case which incorporates a solar panel, too.

I don’t want to go crazy money-wise but think I may be looking at a RAK device, just for the lower power consumption than an ESP32 board in the event that the location has a few cloudy days.

What can people suggest for me?

Your advice is much appreciated!

**Comment by madeofstown** [comment]
I'm really partial to raspberry pi based nodes. On a pi you could even run a VPN to make remote management super simple for you. 

Powering one with a solar panel is going to be hard though. You'll need a decent sized battery (at least 10000mah) and a big panel (at least 15w).

**Reply by stanhamil** [reply]
Are Pi variants very power-efficient? Won’t a Pi also need a LoRA hat?

Getting a decent battery isn’t an issue but I was hoping that the solar panel/the overall size of everything wouldn’t be too intrusive.

**Reply by madeofstown** [reply]
If you are really stuck with solar power then the Pi is not going to be your best option. A raspberry pi IS probably your best option for remote management and updates so that's why I mentioned it. And yes, you would need a LoRa hat, Waveshare makes a basic cheap one.

**Reply by madeofstown** [reply]
For solar powered nodes you're most power efficient option is going to be something with an nRF chip like the RAK Wisblock. The issue with nRF is that it doesn't connect to WiFi and won't be able to keep accurate time without a GPS or RTC nodule or somebody regularly connecting with Bluetooth to update the time. ESP32 uses about 5x more power than nRF but will connect to WiFi and can be run on solar easier than a raspberry pi.

**Reply by stanhamil** [reply]
I was thinking about using solar power so that my family member can have no qualms at all about having something sitting there, drawing power. I guess it's not a deal-breaker if the device has to be plugged into a power supply.

I didn't know the Wisblock didn't support WiFI, so that may be an issue if I definitely want to use MQTT for mesh expansion and remote management.

I have a spare Wireless Paper which I guess I could use...

**Comment by Soupbones** [comment]
If you want a basic no hassle node, get yourself a Heltec T114 node. I set up one inside a Solar light housing that has a 1 watt panel. I swapped out the 1800mAh 18650 battery for a 18650 with larger capacity and its running great as a remote node thats currently screwed right up a tree and actually gets shaded during the day by the leaves of he tree. The Heltec T114 board is very power efficieent. I've had no issues with it on a 1 watt panel which surprised me a lot, as I thought it wouldnt be enough to keep the node running, but this node has had a better run then the other RAK remote node I have which seems to always have issues with the solart charging or voltage drops that cause the firmware to totally reset and lose all its settings, and thats with two 2200mAh 18650 batteries and a 3 watt panel.

**Reply by stanhamil** [reply]
Thanks, that’s good advice.

T114 boards can be picked up for very little money on the usual Chinese sites, but I didn’t think they were compatible with Meshtastic (they don’t appear in the list of supported devices on the web flasher?)

**Reply by Soupbones** [reply]
They do work with Meshtastic and they are on the web flasher as I update mine from there. You just haven't looked in the right place. They are no ESP32 like you might think being Heltec devices. They are down with the RAK devices nRF52 and thats why they are so good on power too. I'm surpised you haven't heard more on the T114 boards. Andy Kirby has tested them extensivly.
![Uploading Screenshot 2024-12-18 201424.jpg…]()

**Reply by madeofstown** [reply]
Just wanted to point out that since the T114 uses an nRF chip it also will not be able to connect to WiFi.

**Reply by stanhamil** [reply]
@Soupbones I’m surprised I missed this on Andy’s channel.

**Comment by mariotti** [comment]
@stanhamil JIC you do not need WiFi for the node administration. You can use the Meshtastic LoRa itself. Check the "new" remote administration.

**Reply by stanhamil** [reply]
I won’t anywhere within range of this node so the remote administration isn’t going to be an option, which is why I’ll probably use my own MQTT server. I won’t have physical access to this node very often at all.

**Reply by mariotti** [reply]
mmm.. interesting question. Would remote administration work via MQTT?
In that case that wold be a mixture of Wifi and LoRa Admin.
I will check this, sadly after holidays.

[OFF TOPIC]
Which MQTT server are you using? Fully own or some free public?
I mean if you have a suggestion? as I will set one up soon too.

**Reply by stanhamil** [reply]
I’ve set up my own MQTT server. I don’t want to bring in loads of traffic by opening up to a public MQTT server!

**Comment by mariotti** [comment]
@stanhamil I have a similar issue. But I could not find yet a perfect solution :( .

 First "sun", I found out that my location is not ideal, due to a specific building in front of it :) Check it in advance. The number of sun hours is important. 

At present I am testing this "TTGO T-BAT 18650 Battery Compartment with CN3065 Solar Power Manager".

I.e. a small device at some other person place.

There is other solutions, even pretty nice, but I cannot afford a battery explosion at others :( 

I do not have a solution. I hope that for my specific project I can find it soon.

**Reply by stanhamil** [reply]
Light isn’t an issue at my chosen location; the elevation, lack of obstructions and sunlight from that side of the building for most of the day make it a great spot. It won’t benefit me directly by having a node there, but it will contribute the network. Online maps show few nodes around that part of southeast London.

**Comment by barryhunter** [comment]
I'd be tempted to say the SenseCAP-Card-Tracker-T1000-E is *close* to ideal. 

Its fully enclosed, ie already has a waterproof enclosure. And VERY compact. (so quite distreet on the window cill!) 

Its nrf based, so has very good battery life, can last days between charges. 

The main drawback, it only has an internal antenna, which no possibility to use an external high gain one. But the internal one seems reasonable, have had contacts 10km+ with it.  

Also connecting 'external' power is only via 'magnetic' connector, which isnt ideal for a remote device (could easily get dislodged in wind) - so I think would have to 'hot glue' or something the connector on. 

I have powered it with a small solar panel, but only in testing. Haven't left it for more than a few days yet (and only inside a conservatory). (I keep using as my daily device, as it so pocketable!) 
... so a small USB output solar panel should in theory keep it powered for long term. Can get waterproof ones. 

(the device has nice battery protection, so if the battery drains, it should easily recover when power/sunlight recovers. 


The only bit, not sure about is wifi for remote admin. Not even tried that, not sure if possible. 

Have notice bluetooth, does work upto about 10m, so *might* even be able to access the device from the ground. 


Ok, it not exactly the cheapest option (and overpowered - dont really need GPS for fixed repeater), but perhaps offset, by requireing minimal extras - ie dont need an encosure. As it lightweight, it could quite literally just be hung on a small nail, or perhaps just double-sided taped to wall :)
... if get a cheap solar panel, that includes a wall mount, the Sensecap could literally be stuck to the back of the panel. Just with the lora anatena part protruding. 
https://www.aliexpress.com/item/1005007586989066.html

**Reply by stanhamil** [reply]
That’s great advice; thank you.

I had a brainwave today and when I get home from work, I’m going to dismantle one of my solar-powered garden lights and bastardise it to incorporate a Heltec Wireless paper I have spare to see if that’s a viable option.

I’ll find a way to attach/glue the device inside and then drill a hole to mount the external cable.

If that initial test works, I may invest in a couple more solar lights and some T114s…

**Reply by barryhunter** [reply]
Yes, garden lights might well be a really nice source of a solar panel, and enclosure. Been on the lookout for a good one!

... would be weary of relying on the device battery and charging circuit. Such lights often have very basic circuitry, and omit 'under voltage' protection, such that keeps outputting a voltage, rather than cutting off cleanly (for LED that might be ok, they just dim). 

... but for powering a microcontroller, that not idea, practically if it power cycles. Have had issues with a Heltec V3, in this regards. Ends up getting currupted occasionally, when battery is drained. Have seen it hwere the device just power cycles a lot. 

... also the lora node, might draw quite a bit of current, when transmitting. Possibly more than a cheap/small battery can provide. 


Suggest pairing the node with a nice battery. And just use the garden light output as 'charging input' (even if it actully has its own battery!) 

No experience if the Wireless Paper, or the T114 had good battery management, but so far been impressed with the Sensecap. Definitly wouldnt trust a heltec v3 for a 'zero maintaince' node (even if had a decent battery+solar panel, to cope with power).

**Reply by stanhamil** [reply]
I have a few V3s and haven’t personally had any problems with them, but I’ve read a lot about various issues.

I’d probably pack in a few 18650 batteries.

I was previously looking at buying the parts separately, but the garden light is a good start.

**Reply by barryhunter** [reply]
Andys video
https://www.youtube.com/watch?v=ROzkbjWR8lw
does talk about including battery protection, when using plain 18650
